HPXMLtoOpenStudio/resources/battery.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+class Battery
+  def self.apply(model, battery)
+    obj_name = battery.id
+
+    default_values = Battery.get_battery_default_values()
+
+    voltage = battery.voltage # V
+    capacity = get_kWh_from_Ah(battery.capacity, voltage) # kWh
+
+    return if capacity <= 0 || voltage <= 0
+
+    is_outside = (battery.location == HPXML::LocationOutside) || (battery.location == HPXML::LocationExteriorWall) || (battery.location == HPXML::LocationRoofDeck)
+    if not is_outside
+      frac_sens = 1.0
+    else # Internal gains outside unit
+      frac_sens = 0.0
+    end
+
+    # The following calculations are from Rohit C.
+    number_of_cells_in_series = Integer((voltage / 3.6).round)
+    number_of_strings_in_parallel = Integer(((capacity * 1000.0) / (voltage * 3.2)).round)
+    battery_mass = (capacity / default_values[:capacity]) * 99.0 # kg
+    battery_surface_area = 0.306 * (capacity**(2.0 / 3.0)) # m^2
+
+    elcds = model.getElectricLoadCenterDistributions
+    if elcds.empty?
+      elcd = OpenStudio::Model::ElectricLoadCenterDistribution.new(model)
+      elcd.setName('Battery elec load center dist')
+      elcd.setElectricalBussType('AlternatingCurrentWithStorage')
+    else
+      elcd = elcds[0]
+
+      if elcd.inverter.is_initialized
+        elcd.setElectricalBussType('DirectCurrentWithInverterDCStorage')
+      end
+    end
+
+    # elcd.setGeneratorOperationSchemeType('TrackElectrical')
+    # elcd.setDemandLimitSchemePurchasedElectricDemandLimit(0)
+    # elcd.setStorageOperationScheme('TrackFacilityElectricDemandStoreExcessOnSite')
+    # elcd.setMaximumStorageStateofChargeFraction(0.95)
+    # elcd.setMinimumStorageStateofChargeFraction(0.20)
```

Comment:
These seem really important. Why are they commented out?

Reply:
These are found in the EnergyPlus testfile, but it's not clear to me if that's what we want to set. Rohit has not sent me an example idf yet.

Reply:
@rchintala13 will look at which control strategies we want to use.

Reply:
Per our discussion:
- Use SAM defaults for min/max SOC
- GeneratorOperationSchemeType and DemandLimitSchemePurchasedElectricDemandLimit don't apply to PV, since PV always generates based on the sun
- StorageOperationScheme should be TrackFacilityElectricDemandStoreExcessOnSite